src/main/java/fi/okm/jod/yksilo/config/SessionConfig.java

```diff
@@ -0,0 +1,105 @@
+import org.springframework.data.redis.connection.RedisConfiguration;
+import org.springframework.data.redis.connection.RedisStandaloneConfiguration;
+import org.springframework.data.redis.connection.lettuce.RedisCredentialsProviderFactory;
+import org.springframework.data.redis.serializer.GenericJackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.lang.NonNull;
+import org.springframework.security.jackson2.SecurityJackson2Modules;
+import org.springframework.session.data.redis.config.annotation.web.http.EnableRedisHttpSession;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain;
+
+@Configuration(proxyBeanMethods = false)
+@EnableRedisHttpSession
+@Slf4j
+public class SessionConfig implements BeanClassLoaderAware {
+  private ClassLoader loader;
+
+  @Override
+  @SuppressFBWarnings(value = {"EI_EXPOSE_REP2"})
+  public void setBeanClassLoader(@NonNull ClassLoader classLoader) {
+    this.loader = classLoader;
+  }
+
+  @Bean
+  public RedisSerializer<Object> springSessionDefaultRedisSerializer() {
+    // Create a custom ObjectMapper that uses Spring Security’s Jackson modules.
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModules(SecurityJackson2Modules.getModules(this.loader));
+    return new GenericJackson2JsonRedisSerializer(mapper);
+  }
+
+  @Bean
+  @Profile("cloud")
+  LettuceClientConfigurationBuilderCustomizer lettuceClientConfigurationBuilderCustomizer() {
+    return builder ->
+        builder.redisCredentialsProviderFactory(
+            new RedisCredentialsProviderFactory() {
+              @Override
+              public RedisCredentialsProvider createCredentialsProvider(
+                  @NonNull RedisConfiguration redisConfiguration) {
+                if (redisConfiguration
+                    instanceof RedisStandaloneConfiguration redisStandaloneConfiguration) {
+                  // Custom implementation of RedisCredentialsProvider for IAM Authentication.
+
+                  // References:
+                  // https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/auth-iam.html#auth-iam-Connecting
+                  // https://github.com/aws-samples/elasticache-iam-auth-demo-app/tree/main
+
+                  // The username is the same as the user id.
+                  String username = redisStandaloneConfiguration.getUsername();
+
+                  // The cache name is the subdomain of the host name without the last part.
+                  String hostName = redisStandaloneConfiguration.getHostName();
+                  String subdomain = hostName.split("\\.")[0];
+                  String cacheName = subdomain.substring(0, subdomain.lastIndexOf("-"));
```

Comment:
Is it guaranteed (e.g. by the reference documentation) that the cache name is a prefix of the host name? Maybe it would be safer to provide it as a separate parameter.

Reply:
I think it is. Cache name can't contains dots, so the subdomain is the cache name + random hash separated by dash. If we add it as a separate parameter, we can't use Spring Boot default configuration for reading all Spring Data Redis parameters.

Reply:
It _looks_ like the endpoint URL is constructed from the cache name in a certain way, but unless that is specified somewhere, I would not rely on that. E.g. https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/Endpoints.html suggests that the format of the URL varies e.g. based on the cache type (stand-alone, stand-alone with encrypted transit, ...).
It seems that with the auto-configuration, one can anyway configure only one Redis instance, so an additional configuration parameter is not a problem.